templates/java/org/prism/Loader.java.erb

```diff
@@ -310,6 +310,18 @@ public class Loader {
         return (short) flags;
     }
 
+    private VariableInteger loadVariableInteger() {
+        boolean negative = buffer.get() != 0;
+        int length = loadVarUInt();
+        int[] nodes = new int[length];
+
+        for (int i = 0; i < length; i++) {
+            nodes[i] = loadVarUInt();
+        }
+
+        return new VariableInteger(negative, nodes);
```

Comment:
I think it would be far more useful if this returned `int`, `long` or `BigInteger` (and so an `Object` return type for this method).

That would also be a lot faster, allocating extra VariableInteger+int[] for the common case where it fits in `int` would be I think a significant overhead.